tensorboard/plugins/histogram/summary_v2.py

```diff
@@ -426,19 +427,29 @@ def _buckets_v3(data, bucket_count=None):
         tf.debugging.assert_type(bucket_count, tf.int32)
         data = tf.reshape(data, shape=[-1])  # flatten
         data = tf.cast(data, tf.float64)
-        is_empty = tf.equal(tf.size(input=data), 0)
+        data_size = tf.size(input=data)
+        is_empty = tf.logical_or(
+            tf.equal(data_size, 0), tf.less_equal(bucket_count, 0)
+        )
 
         def when_empty():
-            return tf.constant([], shape=(0, 3), dtype=tf.float64)
+            """When input data is empty or bucket_count is zero.
+
+            1. If bucket_count is specified as zero, an empty tensor of shape
+              (0, 3) will be returned.
+            2. If the input data is empty, a tensor of shape (bucket_count, 3)
+              of all zero values will be returned.
+            """
+            return tf.zeros((max(0, bucket_count), 3), dtype=tf.float64)
```

Comment:
We should use `tf.math.maximum()` here, rather than Python's `max()`, in order to be compatible with `bucket_count` values that aren't native Python numbers (e.g. a `tf.constant()` instead).

Also I'd recommend just adding a conversion up at the top that does `bucket_count = tf.math.maximum(0, bucket_count)` so we don't have to worry about negative bucket counts in any of the later logic.